packages/p2p/src/components/modal-manager/modal-manager-context-provider.jsx

```diff
@@ -1,27 +1,67 @@
 import React from 'react';
 import { ModalManagerContext } from './modal-manager-context';
+import { useStores } from 'Stores';
+import { isDesktop } from '@deriv/shared';
 
 const ModalManagerContextProvider = props => {
-    const [active_modal, setActiveModal] = React.useState(null);
+    const [active_modal, setActiveModal] = React.useState({});
+    const [previous_modal, setPreviousModal] = React.useState({});
+    // for mobile, modals are stacked and not shown alternatingly one by one
+    const [stacked_modal, setStackedModal] = React.useState({});
     const [is_modal_open, setIsModalOpen] = React.useState(false);
+    const [modal_props, setModalProps] = React.useState(new Map());
+    const { general_store } = useStores();
+
+    const registerModalProps = modals => {
+        if (Array.isArray(modals)) {
+            modals.forEach(modal => setModalProps(modal_props.set(modal.key, modal.props)));
+        } else {
+            setModalProps(modal_props.set(modals.key, modals.props));
+        }
+    };
```

Comment:
What are these for?

Reply:
Is it possible to move all the props to the modal component to get rid of passing it from anywhere? For example in `RatingModal`, instead of passing `is_user_recommended_previously`, we can declare `is_recommended_by_user` in the `RatingModal` component.